File: Java/FindMinimumInRotatedSortedArray/src/solution/SolutionTest.java
package solution;

import static org.junit.Assert.*;
import static org.junit.Assert.assertEquals;

public class SolutionTest {

    @org.junit.Test
    public void testFinMin() throws Exception {
        Solution sol = new Solution();
        //sol.findMin(test1);
        // test single element array
        int[] test2 = {1};
        assertEquals(sol.findMin(test2), 1);
        // test unrotated array
        int[] test3 = {1, 2, 3, 4, 5, 6};
        assertEquals(sol.findMin(test3), 1);
        // test rotate one number
        int[] test4 = {2, 3, 4, 5, 6, 1};
        assertEquals(sol.findMin(test4), 1);
        // test rotate last number
        int[] test5 = {6, 1, 2, 3, 4, 5};
        assertEquals(sol.findMin(test5), 1);
    }
}

File: Scala/src/FindMinimumInRotatedSortedArray/SolutionTest.scala
package FindMinimumInRotatedSortedArray

import org.scalatest.FunSuite

class SolutionTest extends FunSuite {
  val sol = new Solution
  test("Test single element") {
    val num1 = Array(0)
    assert(sol.findMin(num1) == 0)
  }
  test("Test unrotated array") {
    val num2 = Array(1, 2, 3, 4, 5, 6)
    assert(sol.findMin(num2) == 1)
  }
  test("Test rotated array") {
    val num3 = Array(3, 4, 5, 6, 1, 2)
    assert(sol.findMin(num3) == 1)
  }
}
